src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,11 +286,25 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
-
-            if (currentMask != 0)
+            if (Sse2.IsSupported)
             {
-                goto FoundNonAsciiDataInCurrentMask;
+                currentSseMaskOrAdvSimdIndex = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
```

Comment:
>currentSseMaskOrAdvSimdIndex [](start = 16, length = 28)

I remember adding a comment about this variable. I think you should use different variables to represent this value for `SSE2` and `AdvSimd`. It is little confusing and hard to maintain with future changes.

Reply:
I agree, but I think this is a reasonable compromise. More info here where I replied to your question :) https://github.com/dotnet/runtime/pull/39506#discussion_r466607234

Reply:
Ah, somehow I missed it. I have responded to your comment in the same thread.

Reply:
Done, fixed now.